tests/testthat/test-utils.R

```diff
@@ -0,0 +1,45 @@
+test_that("input validation works", {
+  dat_list <- list("a" = 1)
+  dat_missing_column <- data.frame("oldValues" = c(1, 2))
+  dat_no_error <- data.frame(
+    "oldValues" = c("a", "b"),
+    "newValues" = c("c", "d")
+  )
+
+  expect_error(checkRecode(dat_list), "'dat' must be a data.frame.")
+  expect_error(checkRecode(dat_missing_column), "'dat' must contain the column 'newValues'.")
+  expect_no_error(checkRecode(dat_no_error))
+})
+
+
+test_that("contradicting values are detected", {
+  dat_no_duplicates <- data.frame(
+    oldValues = c("Italia", "Wales"),
+    newValues = c("Italy", "Wales")
+  )
+
+  dat_na <- data.frame(
+    oldValues = c(NA, "Wales", "Italia", "Italia", "Wales", "Egyptian"),
+    newValues = c("Italy", "Wales", "Italy", "Italien", "UK", "Egypt")
+  )
+
+  dat_duplicates <- data.frame(
+    oldValues = c("Italia", "Wales", "Italia", "Italia", "Wales", "Egyptian"),
+    newValues = c("Italy", "Wales", "Italy", "Italien", "UK", "Egypt")
+  )
+
+  expect_no_error(checkRecode(dat_no_duplicates))
+
+  expect_error(checkRecode(dat_na), "Please check the NAs in 'oldValues'.")
+
+  error_message <- tryCatch(checkRecode(dat_duplicates), error = function(e) {
```

Comment:
Is there a benefit of catching and saving the error, vs. directly using `expect_error()`?

Reply:
This was a bit annoying because of how the error message was formatted (because it includes a data.frame). So yes, for the way the error message is currently printed this is necessary.